src/reference/asciidoc/zeromq.adoc

```diff
@@ -102,3 +102,113 @@ ZeroMqChannel zeroMqPubSubChannel(ZContext context) {
 }
 ----
 ====
+
+[[zeromq-inbound-channel-adapter]]
+=== ZeroMq Inbound Channel Adapter
+
+The `ZeroMqMessageProducer` is a `MessageProducerSupport` implementation with reactive semantics.
+It constantly reads the data from a ZeroMq socket in no-blocking manner and produces it as an infinite `Flux` which is subscribed by the `FluxMessageChannel` or explicitly in the `start()` if output channel is not reactive.
+When no data on the socket, a `consumeDelay` (defaults to 1 second) is applied before the next read attempt.
+
+Only `SocketType.PAIR`, `SocketType.PULL` and `SocketType.SUB` are supported by the `ZeroMqMessageProducer`.
+This component can connect to the remote socket or bind onto TCP protocol with the provided or random port.
+The actual port can be obtained via `getBoundPort()` after this component is started and ZeroMq socket is bound.
+The socket options (e.g. security or write timeout) can be configured via `setSocketConfigurer(Consumer<ZMQ.Socket> socketConfigurer)` callback.
+
+If `receiveRaw` option is set to `true`, a `ZMsg`, consumed from the socket, is sent as is in the payload of the produced `Message`: it's up to downstream flow to parse and convert `ZMsg` respectively.
+Otherwise an `InboundMessageMapper` is used to convert consumed data into a `Message`.
+If received `ZMsg` is multi-frame, the first one is treated as a `ZeroMqHeaders.TOPIC` header this ZeroMq message was published to.
```

Comment:
```suggestion
Otherwise an `InboundMessageMapper` is used to convert the consumed data into a `Message`.
If the received `ZMsg` is multi-frame, the first frame is treated as the `ZeroMqHeaders.TOPIC` header this ZeroMq message was published to.
```